lib/blockscore/collection.rb

```diff
@@ -1,18 +1,239 @@
 module BlockScore
+  # Collection is a proxy between the parent and the asssociated members
+  # where parent is some instance of a resource
+  #
   class Collection < Array
-    def initialize(target)
-      super()
-      @target = target
+    # @!attribute [r] parent
+    # resource which owns a collection of other resources
+    #
+    # @example
+    #   person.question_sets.parent # => person
+    #
+    # @return [BlockScore::Base] a resource
+    #
+    # @api private
+    attr_reader :parent
+
+    # Sets parent and member_class then registers embedded ids
+    #
+    # @param [BlockScore::Base] parent
+    # @param [Class] class of collection members
+    #
+    # @return [undefined]
+    #
+    # @api private
+    def initialize(parent, member_class)
+      @parent       = parent
+      @member_class = member_class
+      register_parent_data
+    end
```

Comment:
I still think the collection class looks good, but I'm not sure if you saw my comment a day or two ago on hipchat after we worked on this.  Pasting it here:

> [Sep-8 6:53 PM] John Backus: @delmer what if we did something like this

```ruby
module BlockScore
  class Collection
    def initialize(parent)
      @parent = parent
      register_parent_data
    end

    def member_class
      self.class::MEMBER_CLASS
    end

    def default_params
      {
        self.class::PARENT_KEY => parent.id
      }
    end
  end
end

module BlockScore
  class Person
    class QuestionSets < Collection
      MEMBER_CLASS = QuestionSet
      PARENT_KEY = :person_id
    end
  end
end
```